### PR TITLE
Remove DML: COW struct

### PR DESCRIPTION
**Pull request by lickel**
This replaces @Heap with @CopyOnWrite which handles storage copying automatically.
This lets me remove `copyStorage()` and makes repeating myself less painful.

### CODE REVIEW

File: wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    /// When writing to the the property, the property wrapper will
+    /// perform copy-on-write by utilizing `isKnownUniquelyReferenced`
+    /// to make a copy of the underlying storage.
+    public var wrappedValue: Value {
+        get {
+            storage.value
+        }
+
+        set {
+            if isEqual(self.wrappedValue, newValue) {
```
Comment (lickel) [inline wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift:33]: - [ ] remove this bit

File: wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+/// A property wrapper which allows easily implementing Copy-on-Write for value types.
+@propertyWrapper
+public struct CopyOnWrite<Value> {
```
Comment (kyleve) [inline wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift:20]: Noting: This is a copy of the market implementation of copy on write. It's largely worked, so seems fine to move here.

File: wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift (hunk 1 above)
Comment (kyleve) [reply wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift:33]: +1 feel to remove, this was largely an optimization where we saw most "theme setters" setting values to the same value. Assuming things are different with protos, so seems fine to remove.

**Comment by dnkoutso** [comment]
Nit commit typo `Sendable clsoure`

**Review by dnkoutso — APPROVED** [review]

**Pull request merged**